cast/src/starknet_commands/declare.rs

```diff
@@ -14,36 +13,64 @@ use starknet::{
     signers::LocalWallet,
 };
 use std::sync::Arc;
+use std::process::Command;
 
 #[derive(Args)]
 #[command(about = "Declare a contract to starknet", long_about = None)]
 pub struct Declare {
-    /// Path to the sierra compiled contract
-    pub sierra_contract_path: Utf8PathBuf,
-
-    /// Path to the casm compiled contract
-    pub casm_contract_path: Utf8PathBuf,
+    /// contract name
+    pub contract: String,
 
     /// Max fee for the transaction. If not provided, max fee will be automatically estimated
     #[clap(short, long)]
     pub max_fee: Option<u128>,
 }
 
 pub async fn declare(
-    sierra_contract_path: &Utf8PathBuf,
-    casm_contract_path: &Utf8PathBuf,
+    contract: &str,
     max_fee: Option<u128>,
     account: &mut SingleOwnerAccount<&JsonRpcClient<HttpTransport>, LocalWallet>,
 ) -> Result<DeclareTransactionResult> {
+    let _ = Command::new("scarb")
+        .current_dir(std::env::current_dir().expect("failed to obtain current dir"))
+        .arg("build")
+        .output()
+        .expect("Failed to build contracts with Scarb");
+
+    let current_dir = std::env::current_dir().expect("failed to obtain current dir");
+    let paths = std::fs::read_dir(format!("{}/target/dev", current_dir.to_str().unwrap()))
+        .expect("failed to read the file maybe build failed");
+
+    let mut maybe_sierra_contract_path: Option<String> = None;
+    let mut maybe_casm_contract_path: Option<String> = None;
+    for path in paths {
+        let path_str = path
+            .expect("path not resolved properly")
+            .path()
+            .to_str()
+            .expect("failed to convert path to string")
+            .to_string();
+        if path_str.contains(&contract[..]) {
+            if path_str.contains(".sierra.json") {
+                maybe_sierra_contract_path = Some(path_str);
+            } else if path_str.contains(".casm.json") {
+                maybe_casm_contract_path = Some(path_str);
+            }
+        }
+    }
+
+    let sierra_contract_path = maybe_sierra_contract_path.expect(&format!("no sierra found for contract: {}", contract)[..]);
+    let casm_contract_path = maybe_casm_contract_path.expect(&format!("no casm found for contract: {}", contract)[..]);
```

Comment:
Snce we will declare one contract at a time, I wonder if it wouldn't be easier to just try to check if specific files exist, eg in pseudocode:

```
    let file_path = Path::new("{}.sierra.json", &contract);

    if file_path.exists() {
        read
        ...
    } else {
        println!("File does not exist!");
        ...
    }
```

What do you think?

Reply:
the problem is, scarb outputs files like this:
```
declare_test_HelloStarknet.sierra.json
```
The point is, I wanted to avoid assuming the exact name of the output files.

Reply:
hmm, I see. 
If I remember correctly, scarb should be able to return a map of compiled artifacts, so maybe we could use this here instead?

Reply:
I'll take a look 👀

Reply:
You could reuse the code from https://github.com/software-mansion/protostar/pull/2120
I guess it is related to https://github.com/software-mansion/protostar/issues/2141 - it would have to be done after merging these two PRs

Reply:
the same as in https://github.com/software-mansion/protostar/pull/2138#discussion_r1246244782: let's address this separately (issue already linked in the code).

Reply:
sorry, but unless you plan to take on it straight away (or already doing this) then I'd rather not. 

I see this part as a crucial element of scarb integration to be honest, and without using those maps it seems like we have half-baked solution, that has to be re-done from scratch basically.

If you wish you can do it in another PR and set this branch as target, but I don't think this would make things easier.

🙏

Reply:
[take sierra and casm from starknet artifacts](https://github.com/software-mansion/protostar/pull/2138/commits/038d08f2a5fb77cd0fa3baf100a4472e4bbeb231)